lib/charms/prometheus_k8s/v0/prometheus_scrape.py

```diff
@@ -1182,15 +1220,54 @@ def _scrape_metadata(self) -> dict:
         """Generate scrape metadata.
 
         Returns:
-            Scrape configutation metadata for this metrics provider charm.
+            Scrape configuration metadata for this metrics provider charm.
         """
-        metadata = {
-            "model": f"{self._charm.model.name}",
-            "model_uuid": f"{self._charm.model.uuid}",
-            "application": f"{self._charm.model.app.name}",
-            "charm_name": f"{self._charm.meta.name}",
-        }
-        return metadata
+        return self.topology.as_dict()
+
+
+class RuleFilesProvider(Object):
+    """A 'prometheus_scrape' provider class for rule files only.
+
+    The RuleFilesProvider class if for sending rules to prometheus, unlike
+    :class:`MetricsEndpointProvider`, which is used for configuring a scrape target
+     (including rules). This is useful for providing "free standing" rules.
+
+    Args:
+        charm: a charm instance that has the `prometheus_scrape` interface on the "provides" side.
+        relation_name: must match metadata.yaml and have `prometheus_scrape` as interface.
+        dir_path: root directory for the collection of rule files.
+        recursive: whether or not to scan for rule files recursively.
+    """
+
+    def __init__(self, charm: CharmBase, relation_name: str, dir_path: str, recursive=True):
+        super().__init__(charm, relation_name)
+        self._charm = charm
+        self._relation_name = relation_name
+        self.topology = JujuTopology.from_charm(charm)
+        self.dir_path = dir_path
+        self.recursive = recursive
```

Comment:
If these properties are not going to be public, I would prefix them with  `_`

Reply:
added `_` to `recursive`. I wonder if `dir_path` and `topology` make sense as public.